pkg/daemon/update_test.go

```diff
@@ -404,12 +404,6 @@ func TestReconcilableSSH(t *testing.T) {
 	newMcfg = helpers.CreateMachineConfigFromIgnition(newIgnCfg)
 	_, errMsg = reconcilable(oldMcfg, newMcfg)
 	checkIrreconcilableResults(t, "SSH", errMsg)
-
-	//check that empty Users does not cause panic
-	newIgnCfg.Passwd.Users = nil
-	newMcfg = helpers.CreateMachineConfigFromIgnition(newIgnCfg)
-	_, errMsg = reconcilable(oldMcfg, newMcfg)
-	checkIrreconcilableResults(t, "SSH", errMsg)
```

Comment:
Rather than just remove this test, it would be way cooler to have it test the thing we actually want to happen, which is that the empty users actually _is_ reconcilable now. 

It looks like there is conveniently already a `checkReconcilableResults` check function, maybe just leave the test in and swap out the function in that last line?